x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/queries/TextExpansionQueryBuilder.java

```diff
@@ -111,6 +131,9 @@ protected void doXContent(XContentBuilder builder, Params params) throws IOExcep
         builder.startObject(fieldName);
         builder.field(MODEL_TEXT.getPreferredName(), modelText);
         builder.field(MODEL_ID.getPreferredName(), modelId);
+        if (tokenPruningConfig != null) {
+            tokenPruningConfig.toXContent(builder, params);
+        }
```

Comment:
```suggestion
        if (tokenPruningConfig != null) {
            builder.field(PRUNING_CONFIG.getPreferredName(), tokenPruningConfig);
        }
```
`PRUNING_CONFIG` is a parse field related only to [TextExpansionQueryBuilder.java](https://github.com/elastic/elasticsearch/pull/102862/files#diff-de8c851b2c000f65bac4a55033f31ac382629cc42d2da6e4ca8c3f5bac50fa99)

We should not assume everything that has a config like this has this name and we should decouple the field name in this object from the object itself.